MapboxNavigation/TopBannerViewController.swift

```diff
@@ -0,0 +1,433 @@
+import Foundation
+import MapboxCoreNavigation
+import MapboxDirections
+
+
+@objc protocol TopBannerViewControllerDelegate: StatusViewDelegate {
+    @objc optional func topBanner(_ banner: TopBannerViewController, didSwipeInDirection direction: UISwipeGestureRecognizer.Direction)
+
+    @objc optional func topBanner(_ banner: TopBannerViewController, didSelect legIndex: Int, stepIndex: Int, cell: StepTableViewCell)
+    
+    @objc optional func topBanner(_ banner: TopBannerViewController, willDisplayStepsController: StepsViewController)
+    
+    @objc optional func topBanner(_ banner: TopBannerViewController, didDisplayStepsController: StepsViewController)
+    
+    @objc optional func topBanner(_ banner: TopBannerViewController, willDismissStepsController: StepsViewController)
+    
+    @objc optional func topBanner(_ banner: TopBannerViewController, didDismissStepsController: StepsViewController)
+}
+
+@objc open class TopBannerViewController: ContainerViewController, StatusViewDelegate {
+    
+    weak var delegate: TopBannerViewControllerDelegate? = nil {
+        didSet {
+            statusView.delegate = delegate
+        }
+    }
+
+    lazy var topPaddingView: TopBannerView = .forAutoLayout()
+    
+    lazy var stepsContainer: UIView = .forAutoLayout()
+    var stepsViewController: StepsViewController?
+    
+    var routeProgress: RouteProgress?
+    
+    lazy var stepsContainerConstraints: [NSLayoutConstraint] = {
+       let constraints = [
+        stepsContainer.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+        stepsContainer.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ]
+        return constraints
+    }()
+    
+    lazy var stepsContainerShow: [NSLayoutConstraint] = {
```

Comment:
Nit: Rename these properties to `shownStepsContrainerConstraints` and `hiddenStepsContainerConstraints` so they don’t look like methods.